store/src/server/api.rs

```diff
@@ -159,6 +159,32 @@ impl api_server::Api for StoreApi {
         }))
     }
 
+    /// Returns details for public (on-chain) account by id.
+    #[instrument(
+        target = "miden-store",
+        name = "store:get_public_account_details",
+        skip_all,
+        ret(level = "debug"),
+        err
+    )]
+    async fn get_public_account_details(
+        &self,
+        request: tonic::Request<GetPublicAccountDetailsRequest>,
+    ) -> Result<Response<GetPublicAccountDetailsResponse>, Status> {
+        let request = request.into_inner();
+        let account = self
+            .state
+            .get_account_details(
+                request.account_id.ok_or(invalid_argument("Account missing id"))?.into(),
+            )
+            .await
+            .map_err(internal_error)?;
+
+        Ok(Response::new(GetPublicAccountDetailsResponse {
+            details: account.to_bytes(),
+        }))
+    }
```

Comment:
Here, we should probably check if the the request is made for a public account (this can be determined by account ID) and return an error if not.

But also, see overall review comment - maybe we should make this endpoint a bit more general.

Reply:
Finally I renamed it to `GetAccountDetails` and it supports all account types, so it doesn't validate account type.